lib/dpl/provider/s3.rb

```diff
@@ -1,11 +1,16 @@
 require 'json'
+require 'set'
 
 module DPL
   class Provider
     class S3 < Provider
       requires 'aws-sdk', version: '~> 2.0'
       requires 'mime-types', version: '~> 2.0'
 
+      def client
```

Comment:
Is using both `client` and `api` necessary? I can't remember how exactly they differ, and I feel that the code is confusing because of it.

Reply:
it seems the `Aws::S3::Resource` when listing all objects of a bucket calls the `list_objects` function which itself in the docs is mentioned to better use `list_objectes_v2` for new code. so far i can't tell what's actually the difference and from the docs it seems all operations should be supported. if you want i'll drop the `client` again and will try to use the `@api` instead?

Reply:
@5c077yP Yes, let us try that.